src/fragmentizer.rs

```diff
@@ -0,0 +1,306 @@
+use imap_codec::fragmentizer::{FragmentInfo, Fragmentizer, LineEnding, LiteralAnnouncement};
+use pyo3::{exceptions::PyTypeError, prelude::*, types::PyBytes};
+
+use crate::encoded::PyLiteralMode;
+
+/// Python class representing a line ending
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[pyclass(name = "LineEnding", eq)]
+pub(crate) enum PyLineEnding {
+    Lf,
+    CrLf,
+}
+
+impl std::fmt::Display for PyLineEnding {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PyLineEnding::Lf => f.write_str("LineEnding.Lf"),
+            PyLineEnding::CrLf => f.write_str("LineEnding.CrLf"),
+        }
+    }
+}
```

Comment:
Is this necessary? `PyLiteralMode` also implements `Display` and I don't know why.